src/Microsoft.Azure.WebJobs.Host/Bindings/FunctionBindingContext.cs

```diff
@@ -52,6 +53,28 @@ public class FunctionBindingContext
             MethodName = functionDescriptor?.LogName;
         }
 
+        /// <summary>
+        /// Creates a new instance.
+        /// </summary>
+        /// <param name="functionInstanceId">The instance ID of the function being bound to.</param>
+        /// <param name="functionCancellationToken">The <see cref="CancellationToken"/> to use.</param>
+        /// <param name="functionInvocationServices">The user logger.</param>
+        /// <param name="functionDescriptor">Current function being executed. </param>
+        /// /// <param name="functionDescriptor">Current retry count. </param>
+        public FunctionBindingContext(
```

Comment:
Let's just add a new ctor taking parameters IFunctionInstanceEx and CancellationToken. All the values needed come from the function instance. FunctionExecutor can then be simplified to call this overload.

Reply:
Fixed.